g++: Comparison is always true

With some g++ (version 5.3.0) I got this warning:

```
/home/jmlemetayer/development/misc/cpptoml/include/cpptoml.h: In static member function ‘static std::__cxx11::string cpptoml::toml_writer::escape_string(const string&)’:
/home/jmlemetayer/development/misc/cpptoml/include/cpptoml.h:3063:26: warning: comparison is always true due to limited range of data type [-Wtype-limits]
             else if (*it >= 0x0000 && *it <= 0x001f)
                          ^
```

This is raised by this line:

https://github.com/skystrife/cpptoml/blob/2051836a96a25e5a2d5283be7f633a157848f15e/include/cpptoml.h#L3063

--- a/include/cpptoml.h
+++ b/include/cpptoml.h
@@ -3060,7 +3060,7 @@ class toml_writer
             {
                 res += "\\\\";
             }
-            else if (*it >= 0x0000 && *it <= 0x001f)
+            else if ((const uint32_t)*it <= 0x001f)
             {
                 res += "\\u";
                 std::stringstream ss;

